apps/studio/src/App.vue

```diff
@@ -42,194 +33,538 @@
+
+        if (typeof payload === "string") {
+          // Handle legacy format (just themeId)
+          this.applyThemeCSS(payload);
+        } else if (payload && payload.themeId) {
+          // Handle new format with CSS content
+          this.applyThemeCSS(payload.themeId, payload.css, payload.baseTheme);
+        }
+      });
 
-    if (this.url) {
-      try {
-        await this.$store.dispatch('openUrl', this.url)
-      } catch (error) {
-        console.error(error)
-        this.$noty.error(`Error opening ${this.url}: ${error}`)
-        throw error
+      // Also listen for window events
+      window.addEventListener("theme-preview", (event) => {
+        console.log(
+          "Theme preview window event received in App.vue:",
+          event.detail
+        );
+
+        if (event.detail && event.detail.themeId) {
+          this.applyThemeCSS(
+            event.detail.themeId,
+            event.detail.css,
+            event.detail.baseTheme
+          );
+        }
+      });
+
+      const query = querystring.parse(window.location.search, {
+        parseBooleans: true,
+      });
+      if (query) {
+        this.url = query.url || null;
+        this.runningWayland = !!query.runningWayland;
       }
-    }
 
+      this.$nextTick(() => {
+        window.main.isReady();
+        // Set appLoaded to true after a short delay to ensure the app is fully initialized
+        setTimeout(() => {
+          this.appLoaded = true;
+
+          // Re-apply the theme after ThemeManagerModal is loaded
+          if (this.themeValue) {
+            console.log(
+              "Re-applying theme after ThemeManagerModal is loaded:",
+              this.themeValue
+            );
+            if (typeof this.themeValue === "string") {
+              this.applyThemeCSS(this.themeValue);
+            } else if (this.themeValue.themeId) {
+              this.applyThemeCSS(
+                this.themeValue.themeId,
+                this.themeValue.css,
+                this.themeValue.baseTheme
+              );
+            }
+          }
+        }, 1000);
+      });
+
+      if (this.url) {
+        try {
+          await this.$store.dispatch("openUrl", this.url);
+        } catch (error) {
+          console.error(error);
+          this.$noty.error(`Error opening ${this.url}: ${error}`);
+          throw error;
+        }
+      }
+    } catch (error) {
+      console.error("Error in App.vue mounted hook:", error);
+    }
   },
   methods: {
+    applyThemeCSS(themeId, cssContent, baseTheme) {
+      console.log(`App.vue: Applying theme CSS for ${themeId}`);
+
+      // Get the theme from the store
+      const theme = this.$store.getters["themes/allThemes"].find(
+        (t) => t.id === themeId
+      );
+
+      if (!theme) {
+        console.error(`App.vue: Theme not found in store: ${themeId}`);
+        return;
+      }
+
+      // Special handling for built-in themes that have their own SCSS files
+      const baseThemes = [
```

Comment:
Should these themes be migrated to your new system?

Reply:
yes indeed, good catch

Reply:
the original ones now work with the new ones